src/event_handler.rs

```diff
@@ -402,12 +501,66 @@ pub async fn handle_event_logs(
     println!("Both the Jobs and Executors subscription streams have ended!");
 }
 
-// Retry a transaction if nonce issues occur
-fn should_retry(error: &anyhow::Error) -> bool {
-    let error_string = format!("{:?}", error);
-    if error_string.contains("code: -32000") && error_string.contains("nonce") {
-        return true;
+// Function to regularly check a transaction for block confirmation and resend it if not included
+async fn resend_pending_transaction(
+    app_state: Data<AppState>,
+    txn: FunctionCall<Arc<HttpSignerProvider>, HttpSignerProvider, ()>,
+    mut pending_txn_hash: H256,
+    nonce: U256,
+    mut gas_limit: U256,
+    mut gas_price: U256,
+    resend_deadline: u128,
+) {
+    // Calculating resend retries number based on the overall deadline and interval in which to resend pending/dropped txns
+    let mut resend_retries = resend_deadline / (RESEND_TXN_INTERVAL as u128);
+    while resend_retries > 0 {
+        sleep(Duration::from_secs(RESEND_TXN_INTERVAL)).await;
+
+        // Get the transaction receipt for the pending transaction to check if it's still pending or been dropped
+        let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
+        let Ok(txn_receipt) = http_rpc_client
+            .get_transaction_receipt(pending_txn_hash)
+            .await
+        else {
+            resend_retries -= 1;
+            continue;
+        };
+
+        if txn_receipt.is_some() {
+            break;
+        }
+
+        // Update gas limit and price for sending the replacement transaction with the original nonce
+        gas_limit += GAS_LIMIT_BUFFER.into();
+        gas_price += (U256::from(RESEND_GAS_PRICE_INCREMENT_PERCENT) * gas_price) / 100;
+
+        // Update replacement transaction metadata and send it to the common chain, stop resending if failed to do so once
+        let replacement_txn = txn.clone();
+        let replacement_txn = replacement_txn
+            .nonce(nonce)
+            .gas(gas_limit)
+            .gas_price(gas_price);
+        let Ok(pending_txn) = replacement_txn.send().await else {
+            resend_retries = 0;
+            break;
+        };
+
+        pending_txn_hash = pending_txn.tx_hash();
+        resend_retries -= 1;
     }
 
-    false
+    // If the original transaction fails to get included in the common chain, send a dummy transaction that is confirmed to be included and will replace the original nonce transaction
```

Comment:
`send a dummy transaction that is confirmed to be included`
What are the reasons that this transaction is confirmed to be included? 

My view is there isn't any such txn, theoretically.

Reply:
Sending this kind of transaction is a common practice to replace a particular nonce pending/stuck transaction in case of sending multiple txns without waiting for confirmation.  
https://help.myetherwallet.com/en/articles/5461454-canceling-or-replacing-a-transaction-after-it-s-been-sent
https://info.etherscan.com/how-to-cancel-ethereum-pending-transactions/#:~:text=The%20trick%20to%20%E2%80%9Ccancel%E2%80%9D%20your%20pending%20transaction%20is%20by%20replacing%20the%20transaction%20with%20another%200%20ETH%20transaction%20with%20a%20higher%20gas%20fee%20sending%20to%20yourself%20with%20the%20same%20nonce%20as%20the%20pending%20transaction.

Reply:
> https://help.myetherwallet.com/en/articles/5461454-canceling-or-replacing-a-transaction-after-it-s-been-sent

This mentioned 
"There is no guarantee this will work, nor that it will work every time. You may end up adding more stuck transactions if you don’t follow the process carefully, and even then there is no guarantee."